src/pattern.rs

```diff
@@ -91,73 +87,34 @@ impl Pattern {
-    ("region", r"R\d"),
-    ("extended", r"EXTENDED"),
-    ("hardcoded", r"HC"),
-    ("proper", r"PROPER"),
-    ("repack", r"REPACK"),
-    ("container", r"MKV|AVI"),
-    ("widescreen", r"WS"),
-    ("three_d", r"3D"),
-    ("unrated", r"UNRATED"),
-    ("language", r"rus\.eng|US"),
-    ("garbage", r"1400Mb|3rd Nov|((Rip)) "),
-    ("imdb", r"tt\d{7}"),
-];
-
 lazy_static! {
-    static ref PATTERNS: HashMap<&'static str, Pattern> = {
-        let mut bucket = HashMap::new();
-
-        for (name, pattern) in &ALL_RAW_PATTERNS {
-            regex!(bucket, *name, pattern);
-        }
-
-        regex!(
-            bucket,
-            "year",
-            r"(?P<year>(1[89]|20)\d\d)",
-            false,
-            true,
-            true
-        );
-
-        regex!(
-            bucket,
-            "website",
-            r"^(\[ ?([^\]]+?) ?\]) ?",
-            true,
-            false,
-            false
-        );
-
-        bucket
-    };
+    pub static ref PATTERN_SEASON: Pattern = regex!(
+        r"[Ss]?(?P<short>\d+) ?[Eex]|(Season|SEASON)(?:[^\d]|$)(?P<long>\d+)|S(?P<dash>\d+) - \d+"
+    );
+    pub static ref PATTERN_EPISODE: Pattern = regex!(
+        r"[Ee](?P<short>\d+)(?:[^\d]|$)|(Episode|EPISODE)(?:[^\d]|$)(?P<long>\d+)|\d+x(?P<cross>\d+)|S\d+ - (?P<dash>\d+)"
+    );
+    pub static ref PATTERN_RESOLUTION: Pattern = regex!(r"((\d{3,4}p))[^M]");
+    pub static ref PATTERN_QUALITY: Pattern = regex!(
+        r"(?:PPV\.)?[HP]DTV|(?:HD)?CAM|B[rR]Rip|TS|(?:PPV )?WEB-?(DL)?(?: DVDRip)?|H[dD]Rip|DVDRip|DVDRiP|DVDRIP|CamRip|W[EB]B[rR]ip|[Bb]lu[Rr]ay|DvDScr|hdtv"
+    );
+    pub static ref PATTERN_CODEC: Pattern =
+        regex!(r"[Xx][Vv][Ii][Dd]|[Xx]264|[hH]\.?264/?|[Xx]265|[Hh]\.?265|[Hh][Ee][Vv][Cc]?");
+    pub static ref PATTERN_AUDIO: Pattern =
+        regex!(r"MP3|DD5\.?1|Dual[\- ]Audio|LiNE|DTS|AAC(?:\.?2\.0)?|AC3(?:\.5\.1)?");
+    pub static ref PATTERN_GROUP: Pattern = regex!(r"(- ?([^ -]+(?:-=\{[^ -]+-?$)?))$");
+    pub static ref PATTERN_REGION: Pattern = regex!(r"R\d");
+    pub static ref PATTERN_EXTENDED: Pattern = regex!(r"EXTENDED");
+    pub static ref PATTERN_HARDCODED: Pattern = regex!(r"HC");
+    pub static ref PATTERN_PROPER: Pattern = regex!(r"PROPER");
+    pub static ref PATTERN_REPACK: Pattern = regex!(r"REPACK");
+    pub static ref PATTERN_CONTAINER: Pattern = regex!(r"MKV|AVI");
+    pub static ref PATTERN_WIDESCREEN: Pattern = regex!(r"WS");
+    pub static ref PATTERN_THREE_D: Pattern = regex!(r"3D");
+    pub static ref PATTERN_UNRATED: Pattern = regex!(r"UNRATED");
+    pub static ref PATTERN_LANGUAGE: Pattern = regex!(r"rus\.eng|US");
+    pub static ref PATTERN_GARBAGE: Pattern = regex!(r"1400Mb|3rd Nov|((Rip)) ");
+    pub static ref PATTERN_IMDB: Pattern = regex!(r"tt\d{7}");
+    pub static ref PATTERN_YEAR: Pattern = regex!(r"(?P<year>(1[89]|20)\d\d)", false, true, true);
+    pub static ref PATTERN_WEBSITE: Pattern = regex!(r"^(\[ ?([^\]]+?) ?\]) ?", true, false, false);
```

Comment:
There's no need to prepend every constant with `PATTERN` here.

Reply:
Done